op-batcher/batcher/channel_manager.go

```diff
@@ -344,3 +355,26 @@ func l2BlockRefFromBlockAndL1Info(block *types.Block, l1info derive.L1BlockInfo)
 		SequenceNumber: l1info.SequenceNumber,
 	}
 }
+
+// Close closes the current pending channel, if one exists, and prevents the
+// creation of any new channels.
+// This ensures that no new frames will be produced, but there may be any number
+// of pending frames produced before this call which should still be published.
```

Comment:
```suggestion
// This ensures that no new blocks will be added to the channel, but there may be any number
// of frames still produced by calling `OutputFrames()`, which flushes the compression buffer.
// These frames still need to be published.
```

Technically, we might still produce frames from the batches in the compression pipeline. But we don't add any more blocks to the channel builder.